vagrant/ansible/roles/ctdb.setup/tasks/main.yml

```diff
@@ -34,13 +28,34 @@
     permanent: yes
     state: enabled
 
+- name: Reload firewall
+  command: firewall-cmd --reload
+
 - name: Add cluster lock to ctdb.conf
   lineinfile:
     dest: /etc/ctdb/ctdb.conf
     line: "\tcluster lock = /ctdb/lock/cluster.lock"
     regexp: "cluster lock"
     insertafter: '\[cluster\]'
 
+- name: Create /etc/ctdb/nodes
+  copy:
+    content: |
+      {% for item in ctdb_network_private_interfaces -%}
+      {{ item }}
+      {% endfor -%}
+    dest: '/etc/ctdb/nodes'
+    mode: '0644'
+
+- name: Create /etc/ctdb/public_addresses
+  copy:
+    content: |
+      {% for item in ctdb_network_public_interfaces -%}
+      {{ item }}/{{ ctdb_network_public_interface_subnet_mask }} {{ ctdb_network_public_interface_name }}
+      {% endfor -%}
+    dest: '/etc/ctdb/public_addresses'
+    mode: '0644'
+
```

Comment:
Yes. But we cannot depend on it because other backends also require the same configuration. It shouldn't hurt to redo the changes here.

Reply:
Can we only create them if they are not existing to avoid overwrite?

Reply:
I would prefer to not add unnecessary artifacts to main tasks because of specific backends needs. Could we consider using gluster ansible collections just to install gluster itself but keep ctdb and samba installation separated ?

Reply:
That's something we thought of during early stages and somehow didn't pursue it further. May be its time? With few additional tasks added to common YAML I think we can completely avoid the dependency on gluster-ansible-features collection. Are you planning to come up with such a change(of course in a separate PR)?

Reply:
Yes, I can do that. Given that now the XFS backend works, almost everything should already be there to not depend on gluster collections. I'll check it and send a new PR with the required changes.

Reply:
> Yes, I can do that. Given that now the XFS backend works, almost everything should already be there to not depend on gluster collections. I'll check it and send a new PR with the required changes.

https://github.com/samba-in-kubernetes/sit-environment/issues/14